setup.py

```diff
@@ -18,6 +19,13 @@ def get_version():
     return re.search(r"""__version__\s+=\s+(?P<quote>['"])(?P<version>.+?)(?P=quote)""", open('src/pycares/_version.py').read()).group('version')
 
 
+def is_free_threaded_python():
+    cflags = sysconfig.get_config_var("CFLAGS") or ""
+    cppflags = sysconfig.get_config_var("CPPFLAGS") or ""
+    return "--disable-gil" in cflags or "--disable-gil" in cppflags
+
+cffi = "cffi @ git+https://github.com/quansight-labs/cffi" if is_free_threaded_python() else "cffi>=1.5.0"
```

Comment:
This will work but this sort of runtime dependency adjustment is frowned upon these days.

Also I wouldn't advise releasing anything depending on our fork of CFFI until we publish it. It definitely still has issues.